### ISSUE TITLE
Out of Memory Issues still in 2.4

### ISSUE THREAD

**Issue by 9thwall** [NativeScript/NativeScript#3153]
Hi, I was wondering if I still need to be using Fresco since upgrading to 2.4.
I was informed [here ](http://stackoverflow.com/questions/40322575/best-practice-for-loading-images-on-nativescript-angular-ios) that Fresco was no longer needed however I'm still getting out of memory crashes. 

I'm just trying to isolate the problem to an image issue or if it's just bad code on my end.

**Comment by triniwiz** [comment]
I'm facing the same problem

**Comment by tsonevn** [comment]
Hi @9thwall , @triniwiz,
Thank you for reporting this issue.

Could you give us some more info about the cases you have Out of Memory exception for `Android`. It would help if you could give sample project, where the issue could be reproduced and debugged locally. In the meantime, you could also use `nativescript-fresco` plugin for displaying your images for Android.

Regards,
@tsonevn

**Comment by triniwiz** [comment]
@tsonevn i'm using it with a pager that contains about 40 images (a lot) but only 1 - 20 is visible by swiping to the left and each of the images can be flipped to reveal an image behind . In 2.4 i saw an improvement before the pager only rendered about 13 before OOM but now it renders about 20 -22 before OOM to test i would swiped the first 20 images then begin flipping them

**Comment by tsonevn** [comment]
Hi @9thwall, @triniwiz,

Regarding to the last comment. Could you verify the size of the image you are loading in your app.  In case  images with big size has been loaded in the device memory, this could lead to this exception. In this case the solution is to minimize the size of the files. To be able to provide additional information for this issue, please provide sample project, which could be debugged locally or just the files you are loading 

Regards,
@tsonevn

**Comment by triniwiz** [comment]
None of the images are larger than 300kb and the total size of the 40 images are `5.2MB`

**Comment by 9thwall** [comment]
Hi, Sorry about the late response..
Basically, my situation is very similar to triniwiz except I believe maybe my issue has to do with background images. I'm trying to be creative with background images in my app and currently I'm loading them with css.

```
Page {
    background-color: #212121;
    background-image: ~/images/bg.jpg;
    background-repeat: no-repeat;
    background-position: center;
    background-size: cover;
}
```
I have different background images displaying depending on the page.  Each about 95kb. 
Other than that, I have fresco loading images in a RadListView 5kb each.

**Comment by markharding** [comment]
This appears after about 50 images have been loaded. Only 2 are in view at a time. 

```
An uncaught Exception occurred on "Thread-28492" thread.
java.lang.OutOfMemoryError: Failed to allocate a 3686412 byte allocation with 518192 free bytes and 506KB until OOM
at dalvik.system.VMRuntime.newNonMovableArray(Native Method)
at android.graphics.Bitmap.nativeCreate(Native Method)
at android.graphics.Bitmap.createBitmap(Bitmap.java:975)
at android.graphics.Bitmap.createBitmap(Bitmap.java:946)
at android.graphics.Bitmap.createBitmap(Bitmap.java:913)
at com.facebook.imagepipeline.memory.BitmapPool.alloc(BitmapPool.java:55)
at com.facebook.imagepipeline.memory.BitmapPool.alloc(BitmapPool.java:30)
at com.facebook.imagepipeline.memory.BasePool.get(BasePool.java:259)
at com.facebook.imagepipeline.platform.ArtDecoder.decodeStaticImageFromStream(ArtDecoder.java:137)
at com.facebook.imagepipeline.platform.ArtDecoder.decodeJPEGFromEncodedImage(ArtDecoder.java:120)
at com.facebook.imagepipeline.decoder.ImageDecoder.decodeJpeg(ImageDecoder.java:151)
at com.facebook.imagepipeline.decoder.ImageDecoder.decodeImage(ImageDecoder.java:85)
at com.facebook.imagepipeline.producers.DecodeProducer$ProgressiveDecoder.doDecode(DecodeProducer.java:189)
at com.facebook.imagepipeline.producers.DecodeProducer$ProgressiveDecoder.access$200(DecodeProducer.java:97)
at com.facebook.imagepipeline.producers.DecodeProducer$ProgressiveDecoder$1.run(DecodeProducer.java:129)
at com.facebook.imagepipeline.producers.JobScheduler.doJob(JobScheduler.java:207)
at com.facebook.imagepipeline.producers.JobScheduler.access$000(JobScheduler.java:27)
at com.facebook.imagepipeline.producers.JobScheduler$1.run(JobScheduler.java:78)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
at com.facebook.imagepipeline.core.PriorityThreadFactory$1.run(PriorityThreadFactory.java:43)
at java.lang.Thread.run(Thread.java:818)
```

**Comment by SvetoslavTsenov** [comment]
Fixed with tns-core-modules 3.0.0
